server/src/main/kotlin/com/kamelia/jellyfish/rest/file/Routes.kt

```diff
@@ -49,24 +59,45 @@ private fun Route.uploadFileFromToken() = postOrCatch(path = "/upload/token") {
     ))
 }
 
-private fun Route.getPagedFiles() = getOrCatch(path = "/{uuid?}") {
+private fun Route.getFile() = get("/{code}") {
+    val user = jwtOrNull()?.uuid?.let { Users.findById(it) }
+    val code = call.getParam("code")
+
+    FileService.getFile(code, user).ifSuccessOrElse(
+        onSuccess = { (data) ->
+            val file = FileUtils.getOrNull(data!!.ownerId, code)
+            if (file != null) {
+                call.respondFile(file, data.name, data.mimeType)
+            } else {
+                FileService.deleteFileByCodeAsAdmin(code) // remove orphaned file
+                call.respond(QueryResult.notFound())
+            }
```

Comment:
Not sure if this is a good idea. Such a rare event should not be handled automatically, instead we should come up with a report system, or simply count orphaned files discovered.

The goal is to tell admins there is a problem and let them handle it.

Reply:
Noted, agreed, ill replace it with a todo comment ; we shall handle that later when we have admin routes